network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/revoked/NetpVpnAccessRevokedDialogMonitor.kt

```diff
@@ -47,14 +49,19 @@ class NetpVpnAccessRevokedDialogMonitor @Inject constructor(
         super.onActivityResumed(activity)
         coroutineScope.launch(dispatcherProvider.io()) {
             if (shouldShowDialog()) {
+                logcat { "VPN beta ended" }
                 // Resetting here so we don't show this dialog anymore
                 withContext(dispatcherProvider.main()) {
                     betaEndedDialog.show(activity)
                 }
-            } else if (networkProtectionRepository.vpnAccessRevoked) {
+            } else if (netpSubscriptionManager.getVpnStatus().isExpired()) {
+                logcat { "VPN subscription status expired" }
                 withContext(dispatcherProvider.main()) {
-                    accessRevokedDialog.show(activity)
+                    accessRevokedDialog.showOnce(activity)
                 }
+            } else {
+                logcat { "VPN access revoke dialog clear shown state" }
+                accessRevokedDialog.clearIsShown()
```

Comment:
This is to ensure the expired dialog (aka access revoked) shows the next time the account expires